### PR TITLE
New: Hint to validate prefixed CSS properties are listed first

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

Tasks:
* [x] Handle prefixed property names (`-moz-appearance`)
* [x] Handle prefixed property values (`display: -ms-grid`)

### CODE REVIEW

**Comment by antross** [comment]
@webhintio/core This is now ready for review.

File: packages/hint-css-prefix-order/src/meta.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const meta: HintMetadata = {
+    docs: {
+        category: Category.compatibility,
+        description: `Ensure vendor-prefixed versions of a CSS property are listed before the unprefixed version.`,
+        name: 'Prefixed CSS first'
+    },
+    id: 'css-prefix-order',
+    schema: [
+        /*
+         * If you want to allow the user to configure your hint
```
Comment (molant) [inline packages/hint-css-prefix-order/src/meta.ts]: Remove this comment

File: packages/hint-css-prefix-order/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    "formatters": [...],
+    "parsers": [...],
+    "hints": {
+        "css-prefix-order": "error"
+    },
+    ...
+}
+```
+
+## Further Reading
```
Comment (molant) [inline packages/hint-css-prefix-order/README.md]: Remove this section if there's nothing else to link to.

File: packages/hint-css-prefix-order/package.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    "eslint-plugin-typescript": "^0.14.0",
+    "hint": "^4.3.1",
+    "markdownlint-cli": "^0.13.0",
+    "npm-link-check": "^3.0.0",
+    "npm-run-all": "^4.1.5",
+    "nyc": "^13.1.0",
+    "postcss": "^7.0.14",
+    "rimraf": "^2.6.3",
+    "typescript": "^3.3.1",
+    "typescript-eslint-parser": "^22.0.0"
```
Comment (molant) [inline packages/hint-css-prefix-order/package.json]: This package is deprecated. It should be `` (or newer)
```suggestion
    "@typescript-eslint/parser": "1.2.0"
```

I don't know why we have it pinned though so maybe we should open an issue about that.

File: packages/hint-css-prefix-order/package.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    "test": "npm run lint && npm run build && npm run test-only",
+    "test-only": "nyc ava",
+    "test-release": "npm run lint && npm run clean && npm run build:assets && tsc && npm run test-only",
+    "init": "npm install && npm run build",
+    "watch": "npm run build && npm-run-all --parallel -c watch:*",
+    "watch:assets": "npm run build:assets -- -w --no-initial",
+    "watch:test": "ava --watch",
+    "watch:ts": "npm run build:ts -- --watch"
+  },
+  "version": "0.0.1"
```
Comment (molant) [inline packages/hint-css-prefix-order/package.json]:
```suggestion
  "version": "1.0.0"
```

**Comment by antross** [comment]
@utsavized Can you review this PR as well if you have time? Thanks!

File: packages/hint-css-prefix-order/package.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "extends": "../../.nycrc"
+  },
+  "peerDependencies": {
+    "@hint/parser-css": "^2.0.3",
+    "hint": "^4.3.1"
+  },
+  "private": true,
+  "repository": "webhintio/hint",
+  "scripts": {
+    "build": "npm run clean && npm-run-all build:*",
```
Comment (molant) [inline packages/hint-css-prefix-order/package.json]: We no longer do the `clean` on build (and we should update `create-hint` for this)

File: packages/hint-css-prefix-order/package.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "@hint/utils-tests-helpers": "^2.0.3",
+    "@types/debug": "0.0.31",
+    "@types/node": "8.0.14",
+    "ava": "^1.2.0",
+    "cpx": "^1.5.0",
+    "eslint": "^5.12.1",
+    "eslint-plugin-markdown": "^1.0.0",
+    "eslint-plugin-typescript": "^0.14.0",
+    "hint": "^4.3.1",
+    "markdownlint-cli": "^0.13.0",
```
Comment (molant) [inline packages/hint-css-prefix-order/package.json]: We don't need this anymore.

**Comment by antross** [comment]
FYI, going to update this PR shortly to actually turn the hint on in our configurations.

**Comment by molant** [comment]
What about writing for @seaotta and put the right category?


________________________________

**Comment by ststimac** [comment]
What do you want me to write?

**Comment by molant** [comment]
I wanted to write waiting. We already have the icon for pitfalls and this hint should go in that category.

**Comment by antross** [comment]
@molant I could wait, but we'll still need to create the Pitfalls category as well... adding the enum entry is easy, but doesn't that also require some additional work on the website? I would assume so since each category has a description that comes from somewhere on https://webhint.io/docs/user-guide/hints/.

**Comment by molant** [comment]
> doesn't that also require some additional work on the website?

Not really, those pages are generated automatically. We need to add the category in the `enum` and make sure the HTML formatter works OK. I think we can get it done today.

**Comment by molant** [comment]
@antross the pitfall PR is merged. You can update the category 😊

File: packages/hint-css-prefix-order/tests/hint.ts [hunk 7]
```diff
@@ -0,0 +1,143 @@
+import generateHTMLPage from 'hint/dist/src/lib/utils/misc/generate-html-page';
```
[the remark this answers is not in the document]
Comment (antross) [reply packages/hint-css-prefix-order/tests/hint.ts]: Not strictly speaking, but I like aligning with what's common so I'll update the filename.

**Pull request merged**